client/app/bundles/course/assessment/question/programming/containers/OnlineEditor/Cpp/OnlineEditorCppView.jsx

```diff
@@ -0,0 +1,367 @@
+            <TableBody displayRowCheckbox={false}>
+              { newDataFilesRows }
+            </TableBody>
+          </Table>
+        </CardText>
+      </Card>
+    );
+  }
+
+  renderTestCases(header, testCases, type) {
+    const allTestCases = this.props.data.get('test_cases');
+    const numAllTestCases = allTestCases.get('public').size + allTestCases.get('private').size
+      + allTestCases.get('evaluation').size;
+
+    const identifier = this.props.intl.formatMessage(translations.identifierHeader);
+    const expression = this.props.intl.formatMessage(translations.expressionHeader);
+    const expected = this.props.intl.formatMessage(translations.expectedHeader);
+    const hint = this.props.intl.formatMessage(translations.hintHeader);
+
+    const rows = [...testCases.get(type).entries()].map(([index, test]) => (
+      <TestCase
+        key={index}
+        {...{
+          updateTestCase: this.props.actions.updateTestCase,
+          deleteTestCase: this.props.actions.deleteTestCase,
+          isLoading: this.props.isLoading,
+          type,
+          index,
+          test,
+          expression,
+          expected,
+          hint,
+        }}
+      />
+      ));
+
+    return (
+      <Card initiallyExpanded>
+        <CardHeader
+          title={header}
+          textStyle={{ fontWeight: 'bold' }}
+          actAsExpander
+          showExpandableButton
+        />
+        <CardText expandable style={{ padding: 0 }}>
+          <Table selectable={false}>
+            <TableHeader adjustForCheckbox={false} displaySelectAll={false}>
+              <TableRow>
+                <TableHeaderColumn className={styles.deleteButtonCell} />
+                <TableHeaderColumn>{identifier}</TableHeaderColumn>
+                <TableHeaderColumn>{expression}</TableHeaderColumn>
+                <TableHeaderColumn>{expected}</TableHeaderColumn>
+                <TableHeaderColumn>{hint}</TableHeaderColumn>
+              </TableRow>
+            </TableHeader>
+            <TableBody displayRowCheckbox={false}>
+              {rows}
+            </TableBody>
+            <TableFooter adjustForCheckbox={false}>
+              <TableRow>
+                <TableRowColumn colSpan="5" style={{ textAlign: 'center' }}>
+                  <FlatButton
+                    label={this.props.intl.formatMessage(translations.addNewTestButton)}
+                    icon={<i className="fa fa-plus" />}
+                    disabled={this.props.isLoading || numAllTestCases >= MAX_TEST_CASES}
+                    onClick={this.testCaseCreateHandler(type)}
+                  />
+                </TableRowColumn>
+              </TableRow>
+            </TableFooter>
+          </Table>
+        </CardText>
+      </Card>
+    );
+  }
+
+  renderEditorCard(header, subtitle, field) {
+    const value = this.props.data.get(field) || '';
+
+    return (
+      <Card containerStyle={{ paddingBottom: 0 }} initiallyExpanded>
```

Comment:
Would it be better if we abstracted out `EditorCard` into a separate component, and pass in the relevant props to support the different languages? I suppose this would attempt to abstract the implementation of the CodeEditor away from this page.